sky/shell/platform/mojo/platform_view_mojo.cc

```diff
@@ -11,9 +13,59 @@ PlatformView* PlatformView::Create(const Config& config) {
   return new PlatformViewMojo(config);
 }
 
-PlatformViewMojo::PlatformViewMojo(const Config& config) : PlatformView(config) {}
+PlatformViewMojo::PlatformViewMojo(const Config& config)
+  : PlatformView(config), dispatcher_binding_(this) {
+}
+
+PlatformViewMojo::~PlatformViewMojo() {
+}
+
+void PlatformViewMojo::Init(mojo::ApplicationImpl* app) {
+  app->ConnectToService("mojo:native_viewport_service", &viewport_);
+
+  mojo::NativeViewportEventDispatcherPtr ptr;
+  dispatcher_binding_.Bind(GetProxy(&ptr));
+  viewport_->SetEventDispatcher(ptr.Pass());
+
+  mojo::SizePtr size = mojo::Size::New();
+  size->width = 800;
+  size->height = 600;
+
+  viewport_->Create(
+      size.Clone(),
+      mojo::SurfaceConfiguration::New(),
+      base::Bind(&PlatformViewMojo::OnMetricsChanged, base::Unretained(this)));
```

Comment:
if you don't want base you can use a lambda here. it's actually very slightly cheaper

viewport_->Create(
  ...
  [this](mojo::ViewportMetricsPtr metrics) { OnMetricsChanged(metrics.Pass())});

but it's a minor difference